pod/perlfunc.pod

```diff
@@ -4350,6 +4350,17 @@ A L<C<my>|/my VARLIST> declares the listed variables to be local
 more than one variable is listed, the list must be placed in
 parentheses.
 
+Note that with a parenthesised list, L<C<undef>|/undef EXPR> can be used
+as a dummy placeholder, for example to skip assignment of initial
+values:
+
+    my ( undef, $min, $hour ) = localtime;
+
+Redeclaring a variable in the same scope or statement will "shadow" the
+previous declaration, creating a new instance and preventing access to
+the previous one. This is usually undesired and, if warnings are enabled,
+will result in a warning in the C<shadow> category.
```

Comment:
Do we have anything in the test suite that demonstrates a C<shadow> warning being thrown in this situation?

If not, should we?

(Same question applies to line 8589 same file.)

Reply:
There are warning for `my` (and `our`) in _t/lib/warnings/pad_ and a warning for a `state` subroutine in _t/op/lexsub.t_

I'm not sure if there are any tests for `state` variable warnings though. (The warnings are of the form `"state" variable $pizza masks earlier declaration in same scope at temp.pl line 4.` and `grep -r '"state" variable' *` produced no matches.)

I'm happy to add a test or two if they genuinely are missing. Should it/they also go in _t/lib/warnings/pad_?

Reply:
I've pushed a commit that adds `state` tests into _t/lib/warnings/pad_. The locations of each new/changed test are pretty arbitrary, so I'm happy to make changes if anyone spots problems or possible improvements.